lib/chef-dk/command/verify.rb

```diff
@@ -141,6 +147,35 @@ def components
           # ------------
           failures = []
 
+          # fail the verify if we have more than one version of chef-provisioning or any of its drivers.
```

Comment:
I think it can be a hard error, as long as the error message very clearly sets expectations for end-users who might encounter it. IMO, finding defects before we would have shipped them is a far bigger priority than making `chef verify` work on every possible configuration that could be encountered in the wild. I think a warning would just end up unseen in some jenkins log and wouldn't be very valuable to us.